integration/renamecollection_compat_test.go

```diff
@@ -0,0 +1,263 @@
+			compatNSFrom: "." + compatCollection.Name(),
+			targetNSTo:   targetDB.Name() + ".newCollection",
+			compatNSTo:   compatDB.Name() + ".newCollection",
+			resultType:   emptyResult,
+		},
+		"EmptyDBTo": {
+			targetNSFrom: targetDB.Name() + "." + targetCollection.Name(),
+			compatNSFrom: compatDB.Name() + "." + compatCollection.Name(),
+			targetNSTo:   ".newCollection",
+			compatNSTo:   ".newCollection",
+			resultType:   emptyResult,
+		},
+		"EmptyCollectionFrom": {
+			targetNSFrom: targetDB.Name() + ".",
+			compatNSFrom: compatDB.Name() + ".",
+			targetNSTo:   targetDB.Name() + ".newCollection",
+			compatNSTo:   targetDB.Name() + ".newCollection",
+			resultType:   emptyResult,
+		},
+		"EmptyCollectionTo": {
+			targetNSFrom: targetDB.Name() + "." + targetCollection.Name(),
+			compatNSFrom: compatDB.Name() + "." + compatCollection.Name(),
+			targetNSTo:   targetDB.Name() + ".",
+			compatNSTo:   targetDB.Name() + ".",
+			resultType:   emptyResult,
+		},
+		"NonExistentDB": {
+			targetNSFrom: "nonExistentDB." + targetCollection.Name(),
+			compatNSFrom: "nonExistentDB." + compatCollection.Name(),
+			targetNSTo:   "nonExistentDB.newCollection",
+			compatNSTo:   "nonExistentDB.newCollection",
+			resultType:   emptyResult,
+		},
+		"NonExistentCollectionFrom": {
+			targetNSFrom: targetDB.Name() + ".nonExistentCollection",
+			compatNSFrom: compatDB.Name() + ".nonExistentCollection",
+			targetNSTo:   targetDB.Name() + ".newCollection",
+			compatNSTo:   targetDB.Name() + ".newCollection",
+			resultType:   emptyResult,
+		},
+		"CollectionToAlreadyExists": {
+			targetNSFrom: targetDB.Name() + "." + targetCollection.Name(),
+			compatNSFrom: compatDB.Name() + "." + compatCollection.Name(),
+			targetNSTo:   targetDB.Name() + "." + targetCollectionExists.Name(),
+			compatNSTo:   targetDB.Name() + "." + compatCollectionExists.Name(),
+			resultType:   emptyResult,
+		},
+		"SameNamespace": {
+			targetNSFrom: targetDB.Name() + "." + targetCollection.Name(),
+			compatNSFrom: compatDB.Name() + "." + compatCollection.Name(),
+			targetNSTo:   targetDB.Name() + "." + targetCollection.Name(),
+			compatNSTo:   targetDB.Name() + "." + compatCollection.Name(),
+			resultType:   emptyResult,
+		},
+		"InvalidNameTo": {
+			targetNSFrom: targetDB.Name() + "." + targetCollection.Name(),
+			compatNSFrom: compatDB.Name() + "." + compatCollection.Name(),
+			targetNSTo:   targetDB.Name() + ".new$Collection",
+			compatNSTo:   targetDB.Name() + ".new$Collection",
+			resultType:   emptyResult,
+		},
+		"LongNameTo": {
+			targetNSFrom: targetDB.Name() + "." + targetCollection.Name(),
+			compatNSFrom: compatDB.Name() + "." + compatCollection.Name(),
+			targetNSTo:   targetDB.Name() + "." + strings.Repeat("aB", 150),
+			compatNSTo:   targetDB.Name() + "." + strings.Repeat("aB", 150),
+			resultType:   emptyResult,
+			altMessage:   "error with target namespace: Invalid collection name: " + strings.Repeat("aB", 150),
+		},
```

Comment:
Great coverage of tests :hugs: At the moment, some of them are unhappy, let's fix them.